src/main/java/io/kuzzle/sdk/API/Controllers/DocumentController.java

```diff
@@ -593,4 +593,55 @@ public DocumentController(final Kuzzle kuzzle) {
 
     return this.mCreateOrReplace(index, collection, documents, null);
   }
+
+  /**
+   * Deletes documents matching the provided search query.
+   *
+   * @param index
+   * @param collection
+   * @param searchQuery
+   * @param waitForRefresh
+   * @return a CompletableFuture
+   * @throws NotConnectedException
+   * @throws InternalException
+   */
+  public CompletableFuture<ArrayList<String>> deleteByQuery(
+      final String index,
+      final String collection,
+      final ConcurrentHashMap<String, Object> searchQuery,
+      final Boolean waitForRefresh) throws NotConnectedException, InternalException {
+
+    final KuzzleMap query = new KuzzleMap();
+
+    query
+        .put("index", index)
+        .put("collection", collection)
+        .put("controller", "document")
+        .put("action", "deleteByQuery")
+        .put("body", new KuzzleMap().put("query", searchQuery))
```

Comment:
Why do you force the "query" object? Other SDKs don't: https://docs.kuzzle.io/sdk/csharp/2/controllers/document/delete-by-query/

Reply:
Maybe we should do it in others SDK?
Since the `query` object will always be here, we could do it inside the SDK ?
With java, we can benefit from avoiding declaring one more `ConcurrentHashMap`. ?

Reply:
It's a thought for later, I removed it from this PR